### ISSUE TITLE
A better way of supporting legacy CABLE build systems

### ISSUE THREAD

**Issue by SeanBryan51** [CABLE-LSM/benchcab#244]
We currently support building legacy CABLE versions via the [`build_script`](https://benchcab.readthedocs.io/en/stable/user_guide/config_options/#build_script) key, where a path to a build script can be provided for benchcab to execute at the build step. Some issues with this approach are:

1. Scripts may require arguments or flags. It is currently not possible to specify flags to scripts, only the path to the script can be specified.
2. Spatial simulations will require multiple executables to be built per version of CABLE (serial and mpi). To support legacy build systems, we will need a way specify multiple build commands for each executable.

This functionality will be needed as most development versions of CABLE will have different build systems, especially when there are significant developments to the CABLE build system for the main branch (e.g. https://github.com/CABLE-LSM/CABLE/pull/200) .

Note, see [here](https://github.com/CABLE-LSM/benchcab/issues/5) for a relevant discussion around this issue.

**Comment by SeanBryan51** [comment]
We can replace the `build_script` parameter with a new parameter that takes in a shell command (or a series of shell commands by specifying [a multiline string](https://yaml.org/spec/1.2-old/spec.html#id2795688)) run from the root directory of the CABLE repository. For example:

```yaml
build_steps: |
  cd offline
  ./build.sh --some-flag
```

**Comment by SeanBryan51** [comment]
After discussing with @ccarouge, we have decided to add support for spack and to deprecate support for legacy CABLE build systems in benchcab. This adds the requirement that users will only be able to test benchcab on branches that can be built by spack. ACCESS-NRI will help with transitioning legacy build systems to use the latest build system compatible with spack.

Moving to spack will solve existing issues around supporting bespoke build scripts (e.g. #244, #249). This is the best way forward considering support for legacy build systems is a temporary that will be deprecated sooner or later.

Closing this issue.
